addons/api/addon/serializers/host-catalog.js

```diff
@@ -38,19 +39,31 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { isPlugin, compositeType } = snapshot.record;
     const { options } = attribute;
-    // Nested conditionals for readability
-    // This deletes any fields that don't belong to the record type
+
+    // Delete any fields that don't belong to the compositeType
+    if (
+      isPlugin &&
+      json &&
+      options?.for &&
+      !options.for.includes(compositeType) &&
```

Comment:
Does this do anything? Since the composite type will be `azure` or `aws`, not `plugin`.
It seems to me that it should have been fine to just add it to the `fieldsByType` and that would be it?

We could then add the usage of `compositeType` to be handled by `for` in the application serializer as part of a refactor in the future to remove the need for `fieldsByType`